src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -593,6 +839,159 @@ public function addAllowedTypes($option, $allowedTypes)
     }
 
     /**
+     * Adds allowed types for one or more options.
+     *
+     * First argument may be a name or an array of option names.
+     *
+     * You can pass a constant of Options interface as first argument to target
+     * a group of options. Supported values are:
+     *
+     *   - Options::ALL Defined and extra options
+     *   - Options::DEFINED
+     *   - Options::NESTED
+     *
+     * If a nested option name is passed it will apply to all nested options.
+     * You can prevent it by passing Options::NONE as fourth argument.
+     *
+     * Any type for which a corresponding is_<type>() function exists is
+     * acceptable. Additionally, fully-qualified class or interface names may
+     * be passed.
+     *
+     * @param string|string[]|int $optionNames  One or more option names
+     *                                          or Options::ALL
+     * @param string|string[]     $allowedTypes One or more accepted types
+     * @param bool                $replace      Whether to replace previous
+     *                                          value
+     * @param int                 $nested       This method is recursive for
+     *                                          nested options which is the
+     *                                          default. Pass Options::NONE
+     *                                          to change it.
+     *
+     * @return OptionsResolver This instance
+     *
+     * @throws UndefinedOptionsException If an option is undefined
+     * @throws AccessException           If called from a lazy option or normalizer
+     */
```

Comment:
@trousers Please stop blindly posting analysis results in every pending pull request.